### PR TITLE
Added schema validations

### PR DESCRIPTION
**Pull request by bzsurbhi**
#### Description of changes

This PR adds below mentioned schema validations:

- Enum value is defined once
- Type is defined once in a schema
- Index is defined once in a model
- Key exists in related model
- Index exists in related model

Also, fixed the logic of 'Validates that any of the fields passed in are in the parent model'
Old behavior: Check if fields passed match in related model
New behavior: Check if fields passed match in parent model itself

##### CDK / CloudFormation Parameters Changed
None

#### Issue #, if available

N/A

#### Description of how you validated changes
Unit tests added

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sundersc — COMMENTED** [review]
Overall LGTM. Can we change the GraqhQL version on this package to 15.8 to match the version used in other packages?

**Review by sundersc — DISMISSED** [review]

File: packages/amplify-schema-validator/src/validators/enum-is-defined-once.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  EnumTypeDefinitionNode,
+} from 'graphql';
+import { ValidationError } from '../exceptions/validation-error';
+
+/**
+     * Enum is defined once
+     *
+     * @param schema graphql schema
+     * @returns true if a enum is defined once
+     */
```
Comment (alharris-at) [inline packages/amplify-schema-validator/src/validators/enum-is-defined-once.ts:13]: Somewhitespace weirdness on this comment

File: packages/amplify-schema-validator/src/validators/enum-is-defined-once.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  ) as EnumTypeDefinitionNode[];
+
+  enumTypeDefinitions.forEach((enumTypeDefinition) => {
+    const enumName = enumTypeDefinition.name.value;
+    const enumValues = enumTypeDefinition.values;
+    if (!enumValues) {
+      /* istanbul ignore next */
+      return;
+    }
+    console.log('schema definition --- ', enumValues);
```
Comment (alharris-at) [inline packages/amplify-schema-validator/src/validators/enum-is-defined-once.ts:27]: Probably want to pull out this log line.

File: packages/amplify-schema-validator/src/validators/index-exists-in-related-model.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import { InvalidDirectiveError } from '../exceptions/invalid-directive-error';
+import { getObjectWithName } from '../helpers/get-object-with-name';
+import { resolveFieldTypeName } from '../helpers/resolve-field-type-name';
+
+/**
+     * Validates that index exists in the related model
+     *
+     * @param schema graphql schema
+     * @returns true if index exists in the related model
+     */
```
Comment (alharris-at) [inline packages/amplify-schema-validator/src/validators/index-exists-in-related-model.ts:16]: Same whitespace thing here

File: packages/amplify-schema-validator/src/validators/type-is-defined-once.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+} from 'graphql';
+import { ValidationError } from '../exceptions/validation-error';
+
+/**
+ * Type is defined once in the schema
+ *
+ * @param schema graphql schema
+ * @returns true if a type is defined once in the schema
+ */
+export const validatetypeIsDefinedOnce = (schema: DocumentNode): Error[] => {
```
Comment (alharris-at) [inline packages/amplify-schema-validator/src/validators/type-is-defined-once.ts:14]: Probably want to capitalize the 't' in type.

**Review by alharris-at — COMMENTED** [review]
Overall LGTM - just a few nits on comment layout, and there's an errant console.log left in here.

File: packages/amplify-schema-validator/src/validators/enum-is-defined-once.ts (hunk 1 above)
Comment (bzsurbhi) [reply packages/amplify-schema-validator/src/validators/enum-is-defined-once.ts]: Fixed

File: packages/amplify-schema-validator/src/validators/enum-is-defined-once.ts (hunk 2 above)
Comment (bzsurbhi) [reply packages/amplify-schema-validator/src/validators/enum-is-defined-once.ts]: Removed

File: packages/amplify-schema-validator/src/validators/index-exists-in-related-model.ts (hunk 3 above)
Comment (bzsurbhi) [reply packages/amplify-schema-validator/src/validators/index-exists-in-related-model.ts]: Fixed

**Review by alharris-at — APPROVED** [review]

**Review by sundersc — APPROVED** [review]

**Pull request merged**